Tools/CMake/AMReX_Config.cmake

```diff
@@ -155,6 +155,54 @@ function (configure_amrex)
       endif()
    endif()
 
+   #
+   # Setup DPC++ -- for now simply add public compile flags
+   #
+   if (ENABLE_DPCPP)
+
+      target_compile_options( amrex
+         PUBLIC
+         $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CXX_COMPILER_ID:Clang>>:-Wno-error=sycl-strict -fsycl -fsycl-unnamed-lambda>)
+
+      target_link_options(amrex PUBLIC -Wno-error=sycl-strict -fsycl -fsycl-unnamed-lambda -device-math-lib=fp32,fp64)
+
+      if (ENABLE_DPCPP_AOT)
+         if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+            ## TODO: use file(READ)
+            execute_process( COMMAND cat /sys/devices/cpu/caps/pmu_name OUTPUT_VARIABLE _cpu_long_name )
+         else ()
+            message(FATAL_ERROR "\nENABLE_DPCPP_AOT is not supported on ${CMAKE_SYSTEM_NAME}\n")
+         endif ()
+
+         string(STRIP "${_cpu_long_name}" _cpu_long_name)
```

Comment:
you could make this an advanced cached var to overwrite the logic from the cmd line

Reply:
Which variable are you referring to? _cpu_long_name

Reply:
Yes, thinking the following situation:
- new CPU on an experimental system with a not-yet-updated AMReX lookup table
- testing with some generic GPU name

Reply:
We probably want to suggest to Intel for their compiler to take the long name from `pmu_name`.  Then we don't need to have that table.

Another issue is in the future, the login and compute nodes might have different cpus.

Reply:
Currently AOT is not useful to us because it does not support device math functions.

Reply:
Sounds great, can one of you please open an issue at https://github.com/intel/llvm for the flags and let's issue a fatal error if AOT is selected in our build logic, because we know it cannot work yet.

Reply:
I commented out the AOT support for now and added an error message if ENABLE_AOT is on. We will re-enable AOT support when the time comes.